src/svc/reconnect.rs

```diff
@@ -0,0 +1,122 @@
+use std::fmt;
+
+use futures::{task, Async, Future, Poll};
+use tower_reconnect;
+
+use super::{NewService, Service};
+
+/// Wraps `tower_reconnect`, handling errors.
+///
+/// Ensures that the underlying service is ready and, if the underlying service
+/// fails to become ready, rebuilds the inner stack.
+pub struct Reconnect<T, N>
+where
+    T: fmt::Debug,
+    N: NewService,
+{
+    inner: tower_reconnect::Reconnect<N>,
+
+    /// The target, used for debug logging.
+    target: T,
+
+    /// Prevents logging repeated connect errors.
+    ///
+    /// Set back to false after a connect succeeds, to log about future errors.
+    mute_connect_error_log: bool,
+}
+
+pub struct ResponseFuture<N: NewService> {
+    inner: <tower_reconnect::Reconnect<N> as Service>::Future,
+}
+
+// ===== impl Reconnect =====
+
+impl<T, N> Reconnect<T, N>
+where
+    T: fmt::Debug,
+    N: NewService,
+    N::InitError: fmt::Display,
```

Comment:
TIOLI: Perhaps this bound should be on the struct as well, since the struct has the `T: fmt::Debug` and `N: NewService` bounds already?